scripts/Muon/GUI/ElementalAnalysis2/auto_widget/ea_auto_tab_view.py

```diff
@@ -0,0 +1,229 @@
+            parameters["threshold"] = float(self.threshold_line_edit.text())
+            parameters["default_width"] = True
+            if not self.default_peak_checkbox.isChecked():
+                parameters["default_width"] = False
+                parameters["min_width"] = float(self.min_width_line_edit.text())
+                parameters["max_width"] = float(self.max_width_line_edit.text())
+                parameters["estimate_width"] = float(self.estimate_width_line_edit.text())
+                if parameters["min_width"] > parameters["max_width"] and \
+                   parameters["min_width"] > parameters["estimate_width"] > parameters["max_width"]:
+                    raise ValueError()
+            if parameters["min_energy"] > parameters["max_energy"]:
+                raise ValueError()
+
+        except ValueError:
+            message_box.warning("ERROR: Invalid arguments for peak finding", self)
+            return None
+        workspace_name = self.find_peaks_combobox.currentText()
+        if workspace_name == "":
+            message_box.warning("ERROR: No workspace selected")
+            return None
+        parameters["workspace"] = workspace_name
+        parameters["plot_peaks"] = self.plot_peaks_checkbox.isChecked()
+
+        return parameters
+
+    def add_options_to_find_peak_combobox(self, options):
+        self.find_peaks_combobox.clear()
+        self.find_peaks_combobox.addItems(options)
+
+    def add_options_to_show_peak_combobox(self, options):
+        self.show_peaks_table_combobox.clear()
+        self.show_peaks_table_combobox.addItems(options)
+
+    def add_options_to_show_matches_combobox(self, options):
+        self.show_match_table_combobox.clear()
+        self.show_match_table_combobox.addItems(options)
+
+    def setup_intial_parameters(self):
+        self.min_energy_line_edit.setText("50")
+        self.max_energy_line_edit.setText("1000")
+        self.threshold_line_edit.setText("0.01")
+        self.min_width_line_edit.setText("0.5")
+        self.max_width_line_edit.setText("30")
+        self.estimate_width_line_edit.setText("3")
+        self.default_peak_checkbox.setChecked(True)
+        self.set_peak_width_visible(False)
+
+    def set_peak_width_visible(self, arg):
+        self.min_width_label.setVisible(arg)
+        self.min_width_line_edit.setVisible(arg)
+        self.max_width_label.setVisible(arg)
+        self.max_width_line_edit.setVisible(arg)
+        self.estimate_width_label.setVisible(arg)
+        self.estimate_width_line_edit.setVisible(arg)
+
+    def enable(self):
+        self.setEnabled(True)
+
+    def disable(self):
+        self.setEnabled(False)
+
+    def set_peak_info(self, workspace, number_of_peaks):
+        if number_of_peaks == 1:
+            label_text = f"{number_of_peaks} peak found in {workspace}"
+        else:
+            label_text = f"{number_of_peaks} peaks found in {workspace}"
```

Comment:
There is some good thinking here. We could probably make this shorter by just doing `peak(s)`
```suggestion
        label_text = f"{number_of_peaks} peak(s) found in {workspace}"
```